service/history/queues/executable.go

```diff
@@ -221,6 +221,12 @@ func (e *executableImpl) Execute() (retErr error) {
 		priorityTaggedProvider := e.taggedMetricsHandler.WithTags(metrics.TaskPriorityTag(e.priority.String()))
 		priorityTaggedProvider.Counter(metrics.TaskRequests.GetMetricName()).Record(1)
 		priorityTaggedProvider.Timer(metrics.TaskScheduleLatency.GetMetricName()).Record(e.scheduleLatency)
+
+		if retErr == nil {
+			e.inMemoryNoUserLatency += e.scheduleLatency + e.attemptNoUserLatency
+		}
+		// if retErr is not nil, HandleErr will take care of the inMemoryNoUserLatency calculation
+		// Not doing it here as for certain errors latency for the attempt should not be counted
```

Comment:
can you call HandleError with nil and handle this in there?

Reply:
That's my initial thought actually. But then I think may not be ideal since with today's Task interface definition HandleError should only be called when there's an error. If we always invoke methods like HandlerError(Execute()), there's no reason why there should be two methods.

yes we can remove the handleError() method totally and let Execute() do everything. But that's out of the scope of this fix I think.

Also the existing fix is actually more robust, whether caller call HandlerError() or not when error is nil don't affect the latency calculation. so i'd say I prefer the current fix.

Let me know if you have strong opinion.